### PR TITLE
Added HttpHeadAttribute

### PR DESCRIPTION
**Pull request by ajaybhargavb**
Issue - #2700

### CODE REVIEW

**Comment by dougbu** [comment]
Good so far but these attributes are also tested in `DefaultActionSelectorTests`, `HttpMethodProviderAttributesTests`, and `RouteTemplateProviderAttributesTest`. Should add `[HttpHead]` there as well.

In addition the `DefaultApplicationModelProviderTest` cases should include use of the `template` constructor parameter.

**Comment by dougbu** [comment]
:watch:

File: test/Microsoft.AspNet.Mvc.Core.Test/DefaultActionSelectorTests.cs [hunk 1]
```diff
@@ -873,22 +874,6 @@ public void RPCMethodWithHttpGet()
             }
         }
 
-        private class HttpMethodAttributeTests_DefaultMethodValidationController
```
Comment (ajaybhargavb) [inline test/Microsoft.AspNet.Mvc.Core.Test/DefaultActionSelectorTests.cs]: Removed because it was unused.

**Comment by ajaybhargavb** [comment]
Updated.

**Comment by dougbu** [comment]
:shipit:

**Pull request merged**